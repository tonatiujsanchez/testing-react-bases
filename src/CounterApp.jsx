import { useState } from "react"

import PropTypes from "prop-types"


export const CounterApp = ({ value }) => {
    
    const [ counter, setCounter ] = useState( value )

    const handleAdd = ( worth ) => {
        setCounter( counter + worth )
    }



    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            
            <button
                onClick={ ()=> handleAdd( -1 )} 
                className="btn"
                style={{
                    marginRight: '15px'
                }}
            >
                -1
            </button>
            <button
                onClick={ ()=> handleAdd( 1 )} 
                className="btn"
                style={{
                    marginRight: '15px'
                }}
            >
                +1
            </button>

            <button
                aria-label='btn-reset'
                onClick={ ()=> setCounter(value) } 
                className="btn"
            >
                Reset
            </button>
        </>
    )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
