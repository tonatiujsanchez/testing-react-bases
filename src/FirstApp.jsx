import PropTypes from 'prop-types'



export const FirstApp = ({ title, subtitle, name }) => {
    
    return (
        <div>
            <h1 data-testid="test-title" style={{ margin: 0 }}> { title } </h1>
            <h2 style={{ margin: 0 }}><strong>{ subtitle }</strong></h2>
            <h2 style={{ margin: 0 }}><strong>{ subtitle }</strong></h2>
            <p style={{ margin: 0 }}>{ name }</p>
        </div>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name: 'Tonatiuj Sácnhez',
    subtitle: 'Sin subtítulo',
    // title: 'Sin título',
}
