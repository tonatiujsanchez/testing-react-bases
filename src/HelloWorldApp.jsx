

export const HelloWorldApp = () => {
    return (
        <h1>HelloWorld App!!</h1>
    )
}
