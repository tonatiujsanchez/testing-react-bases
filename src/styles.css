


html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

.btn {
    padding: 5px 10px;
    font-size: 1rem;
    font-weight: bold;
    cursor: pointer;
}