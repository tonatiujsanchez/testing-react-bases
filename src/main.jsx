import React from 'react'
import ReactDOM from 'react-dom/client'


// import { HelloWorldApp } from './HelloWorldApp'
// import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'

import './styles.css'


ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <FirstApp title="Hi! i'm Ton.dev" /> */}
        <CounterApp value={ 100 } />
    </React.StrictMode>
)
