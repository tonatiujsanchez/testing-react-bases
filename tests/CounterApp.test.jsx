import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from '../src/CounterApp'

describe('Pruebas en el componente CounterApp', () => {
    
    const initialValue = 100
    
    test('Debe de hace match con el snapshot', () => {

        const { container } = render( <CounterApp value={initialValue} /> )
        expect( container ).toMatchSnapshot()

    })

    test('Debe de mostrar el valor inicial de 100', () => {
        

        render( <CounterApp value={initialValue} /> )
        
        expect( screen.getByText( initialValue ) ).toBeTruthy()
        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain( '100' )

    })


    test('Debe de incrementar con el boton +1', () => {

        render( <CounterApp value={initialValue} /> )

        fireEvent.click( screen.getByText('+1') )

        expect( screen.getByRole('heading',{ level: 2 }).innerHTML ).toContain('101')

    })


    test('Debe de decremetar con el boton -1', () => {
        render( <CounterApp value={ initialValue } /> )
        
        fireEvent.click( screen.getByText('-1') )

        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain('99')
    })

    test('Debe de funcionae el boton de reset', () => {

        render( <CounterApp value={ 100 } /> )

        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )

        // fireEvent.click( screen.getByText('Reset') )
        fireEvent.click( screen.getByRole('button',{ name: 'btn-reset' }) )
        
        screen.debug()

        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain('100')


    })



})