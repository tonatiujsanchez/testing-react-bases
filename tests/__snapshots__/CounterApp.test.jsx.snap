// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CounterApp Debe de hace match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    class="btn"
    style="margin-right: 15px;"
  >
    -1
  </button>
  <button
    class="btn"
    style="margin-right: 15px;"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
    class="btn"
  >
    Reset
  </button>
</div>
`;
